packages/lodestar/src/network/gossip/validation/aggregateAndProof.ts

```diff
@@ -75,72 +75,92 @@ export function hasAttestationParticipants(attestation: Attestation): boolean {
-    ) {
-      logger.warn("Rejected gossip aggregate and proof", {reason: "invalid signature", ...logContext});
-      return ExtendedValidatorResult.reject;
-    }
+    committee = epochCtx.getBeaconCommittee(attestation.data.slot, attestation.data.index);
+  } catch (error) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_AGGREGATOR_NOT_IN_COMMITTEE,
+      ...logContext,
+      aggregatorIndex: aggregateAndProof.message.aggregatorIndex,
+      job: attestationJob,
+    });
+  }
+  if (!committee.includes(aggregateAndProof.message.aggregatorIndex)) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_AGGREGATOR_NOT_IN_COMMITTEE,
+      ...logContext,
+      aggregatorIndex: aggregateAndProof.message.aggregatorIndex,
+      job: attestationJob,
+    });
+  }
+  if (!isAggregatorFromCommitteeLength(config, committee.length, aggregateAndProof.message.selectionProof)) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_INVALID_AGGREGATOR,
+      ...logContext,
+      job: attestationJob,
+    });
+  }
+  const aggregator = epochCtx.index2pubkey[aggregateAndProof.message.aggregatorIndex];
+  if (
+    !isValidSelectionProofSignature(
+      config,
+      state,
+      attestation.data.slot,
+      aggregator,
+      Signature.fromCompressedBytes(aggregateAndProof.message.selectionProof.valueOf() as Uint8Array)
+    )
+  ) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_INVALID_SELECTION_PROOF,
+      ...logContext,
+      aggregatorIndex: aggregateAndProof.message.aggregatorIndex,
+      job: attestationJob,
+    });
+  }
+  if (
+    !isValidAggregateAndProofSignature(
+      config,
+      state,
+      computeEpochAtSlot(config, attestation.data.slot),
+      aggregator,
+      aggregateAndProof
+    )
+  ) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_INVALID_SIGNATURE,
+      ...logContext,
+      job: attestationJob,
+    });
+  }
 
-    // TODO: once we have pool, check if aggregate block is seen and has target as ancestor
+  // TODO: once we have pool, check if aggregate block is seen and has target as ancestor
 
-    if (!isValidIndexedAttestation(epochCtx, state, epochCtx.getIndexedAttestation(attestation))) {
-      logger.warn("Rejected gossip aggregate and proof", {reason: "invalid indexed attestation", ...logContext});
-      return ExtendedValidatorResult.reject;
-    }
-  } catch (e) {
-    // if any errors are thrown in the process of checking for REJECTs, return a REJECT
-    logger.warn("Rejected gossip aggregate and proof", {reason: e.message, ...logContext});
-    return ExtendedValidatorResult.reject;
+  if (!isValidIndexedAttestation(epochCtx, state, epochCtx.getIndexedAttestation(attestation))) {
+    throw new AttestationError({
+      code: AttestationErrorCode.ERR_INVALID_INDEXED_ATTESTATION,
+      ...logContext,
+      job: attestationJob,
+    });
   }
-
-  return ExtendedValidatorResult.accept;
```

Comment:
seems like we are missing log that we received valid aggregate and proof gossip

Reply:
it's up here:
https://github.com/ChainSafe/lodestar/pull/1658/files#diff-bea8a0391010a5d078090892ebe28da65cfb8b074c55a935136ea65b7f974a78R68